Framework/DataHandling/src/LoadRawHelper.cpp

```diff
@@ -40,12 +40,15 @@ using Types::Core::DateAndTime;
 
 /// Constructor
 LoadRawHelper::LoadRawHelper()
-    : isisRaw(new ISISRAW2), m_list(false), m_interval(false), m_spec_list(),
-      m_spec_min(0), m_spec_max(EMPTY_INT()), m_numberOfPeriods(0),
+    : m_list(false), m_interval(false), m_spec_list(), m_spec_min(0),
+      m_spec_max(EMPTY_INT()), m_numberOfPeriods(0), m_isis_raw(),
       m_cache_options(), m_specTimeRegimes(), m_prog(0.0), m_numberOfSpectra(0),
       m_monitordetectorList(), m_bmspeclist(false), m_total_specs(0),
       m_logCreator() {}
 
+/// Destructor
+LoadRawHelper::~LoadRawHelper() {}
```

Comment:
Do we need an explicit destructor?
I'm pretty sure if we do
`~LoadRawHelper() = default` in the header the compiler will automatically handle the `std::unique_ptr` for us

Reply:
So the issue here is that the `unique_ptr` wraps a forward-declared type so in the header the size is unknown. Leaving the compiler to generate a default or explicitly requesting one in the header leads to:
```
usr/include/c++/5/bits/unique_ptr.h: In instantiation of ‘void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = ISISRAW2]’:
/usr/include/c++/5/bits/unique_ptr.h:236:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = ISISRAW2; _Dp = std::default_delete<ISISRAW2>]’
../../Framework/DataHandling/inc/MantidDataHandling/LoadRawHelper.h:57:3:   required from here
/usr/include/c++/5/bits/unique_ptr.h:74:22: error: invalid application of ‘sizeof’ to incomplete type ‘ISISRAW2’
  static_assert(sizeof(_Tp)>0,
```
because it is trying to call `delete` on a type whose size is unknown. In hindsight I think I might just remove the forward declaration instead. This header is a leaf header so will not be included by many things.

Reply:
Actually I can't do that as the header is kept private and is in the src folder so we'll have to live with this I think.